### PR TITLE
Update tests to use new API where easily possible

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
The primary thing that wasn't updated yet is the compile and compile_with... functions.

### CODE REVIEW

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try

File: lib/api/src/lib.rs [hunk 1]
```diff
@@ -58,23 +62,214 @@ pub mod memory {
...
+
+            #[cfg(any(feature = "default-backend-cranelift", feature = "docs"))]
+            return Backend::Cranelift;
+
+            #[cfg(all(feature = "default-backend-llvm", not(feature = "docs")))]
+            return Backend::LLVM;
+        }
+    }
+
+    impl std::str::FromStr for Backend {
```
Comment (syrusakbary) [inline lib/api/src/lib.rs:152]: Nice touch!

Comment (MarkMcCaskey) [reply lib/api/src/lib.rs:152]: It's copied and pasted from `wasmer_runtime`, but I agree!  It's very nice!

**Pull request merged**